### PR TITLE
Refactor TRPO and VPG with EpisodesSampler

### PR DESCRIPTION
**Pull request by HenriDeh**
PR Checklist

- [x] Update NEWS.md?
- [x] Unit tests for all structs / functions?
- [x] Integration and correctness tests using a simple env?
- [x] PR Review?
- [x] Add or update documentation?
- [x] Write docstrings for new methods?

### CODE REVIEW

**Comment by HenriDeh** [comment]
CI can't pass without 3.3 released

**Comment by HenriDeh** [comment]
I can't get buildkite to rerun now that Traj 0.3.3 is released. But tests pass. Note that this PR does not make any promise on the correctness  of the two algorithms.

**Comment by HenriDeh** [comment]
Found some issues. Do not review yet.

**Comment by HenriDeh** [comment]
I concluded that the algorithms are broken but due to additional reasons that are not tied to the refactor. Hence, I kept the algorithms commented but at least the optimise! functions are refactored and may serve as an example for other refactors in the future. It's all I can do in this PR. Further fixing of policy gradient algorithm require contributors knowledgeable about how they work.

**Review by jeremiahpslewis — DISMISSED** [review]

**Comment by HenriDeh** [comment]
need a new approval due to conflicts

File: src/ReinforcementLearningCore/Project.toml [hunk 1]
```diff
@@ -42,7 +43,7 @@ Parsers = "2"
 ProgressMeter = "1"
 Reexport = "1"
 ReinforcementLearningBase = "0.12"
-ReinforcementLearningTrajectories = "^0.3.2"
+ReinforcementLearningTrajectories = "0.3.3"
```
Comment (jeremiahpslewis) [inline src/ReinforcementLearningCore/Project.toml:46]:
```suggestion
ReinforcementLearningTrajectories = "^0.3.3"
```
I think we want [0.3.3,0.4), right?

File: src/ReinforcementLearningExperiments/Project.toml [hunk 2]
```diff
@@ -22,7 +23,7 @@ Reexport = "1"
 ReinforcementLearningBase = "0.12"
 ReinforcementLearningCore = "0.12, 0.13"
 ReinforcementLearningEnvironments = "0.8"
-ReinforcementLearningZoo = "0.7, 0.8"
+ReinforcementLearningZoo = "0.8.3"
```
Comment (jeremiahpslewis) [inline src/ReinforcementLearningExperiments/Project.toml:26]:
```suggestion
ReinforcementLearningZoo = "^0.8.3"
```
Same as above

File: src/ReinforcementLearningZoo/Project.toml [hunk 3]
```diff
@@ -28,7 +29,7 @@ LogExpFunctions = "0.3"
 NNlib = "0.8, 0.9"
 Optim = "1"
 ReinforcementLearningBase = "0.12"
-ReinforcementLearningCore = "0.12, 0.13"
+ReinforcementLearningCore = "0.12.3, 0.13"
```
Comment (jeremiahpslewis) [inline src/ReinforcementLearningZoo/Project.toml:32]:
```suggestion
ReinforcementLearningCore = "^0.12.3, 0.13"
```

**Review by jeremiahpslewis — APPROVED** [review]

**Review by jeremiahpslewis — CHANGES_REQUESTED** [review]
see above

**Review by jeremiahpslewis — APPROVED** [review]

**Pull request merged**